gradle-plugin/kubernetes/src/main/java/org/eclipse/jkube/gradle/plugin/task/KubernetesPushTask.java

```diff
@@ -0,0 +1,67 @@
+/**
+ * Copyright (c) 2019 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at:
+ *
+ *     https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.jkube.gradle.plugin.task;
+
+import org.eclipse.jkube.gradle.plugin.KubernetesExtension;
+import org.eclipse.jkube.kit.build.service.docker.config.handler.ImageConfigResolver;
+import org.eclipse.jkube.kit.common.RegistryConfig;
+import org.eclipse.jkube.kit.config.service.BuildServiceConfig;
+import org.eclipse.jkube.kit.config.service.JKubeServiceException;
+import org.eclipse.jkube.kit.config.service.JKubeServiceHub;
+
+import javax.inject.Inject;
+import java.io.IOException;
+import java.util.Collections;
+
+public class KubernetesPushTask extends AbstractJKubeTask {
+  @Inject
+  public KubernetesPushTask(Class<? extends KubernetesExtension> extensionClass) {
+    super(extensionClass);
+    setDescription("Uploads the built Docker images to a Docker registry");
+  }
+
+
+  @Override
+  protected JKubeServiceHub.JKubeServiceHubBuilder initJKubeServiceHubBuilder() {
+    return TaskUtil.addDockerServiceHubToJKubeServiceHubBuilder(super.initJKubeServiceHubBuilder(), kubernetesExtension, javaProject, kitLogger)
+      .buildServiceConfig(buildServiceConfigBuilder().build());
+  }
+
+  @Override
+  public void run() {
+    if (kubernetesExtension.getSkipPushOrDefault()) {
+      return;
+    }
+
+    ImageConfigResolver imageConfigResolver = new ImageConfigResolver();
```

Comment:
Why aren't we using the `resolvedImages` field provided by AbstractJKubeTask like in the rest?